cpp_faster_fifo/cpp_lib/faster_fifo.cpp

```diff
@@ -162,14 +172,18 @@ bool timer_positive(const struct timeval &timer) {
 int queue_put(void *queue_obj, void *buffer, const void **msgs_data, const size_t *msg_sizes, const size_t num_msgs, const int block, const float timeout) {
     auto q = (Queue *)queue_obj;
     LockGuard lock(&q->mutex);
+    if (get_queue_size(q) >= q->get_max_size()) {
```

Comment:
I don't think this is the right fix.
If the queue already has a max number of elements, we ought to wait for some elements to be removed for the specified timeout time.

This logic is implemented in the while loop below and in the can_fit function you modified. So I think this if statement needs to be removed.